### PR TITLE
allow blank lines in csv files imported into applab.

### PR DESCRIPTION
**Pull request by davidsbailey**

### CODE REVIEW

**Comment by laurelfan** [comment]
laurelbot says: test?

File: shared/middleware/tables_api.rb [hunk 1]
```diff
@@ -122,7 +122,7 @@ class TablesApi < Sinatra::Base
               "contains a name for each column:<br><br>#{columns.join(',')}"
         halt 400, {}, msg unless column
       end
-      CSV.foreach(tempfile, headers: true) do |row|
+      CSV.foreach(tempfile, headers: true, skip_blanks: true) do |row|
```
Comment (islemaster) [inline shared/middleware/tables_api.rb]: Trying to understand this reviewing the Ruby [CSV documentation](http://ruby-doc.org/stdlib-1.9.2/libdoc/csv/rdoc/CSV.html).  I find it awesome and very confusing that ruby understands here that `headers: true, skip_blanks: true` are not arguments 2 and 3, but are both part of a hash that makes up argument 2.  But this does [conform to the style guide](https://github.com/bbatsov/ruby-style-guide#no-braces-opts-hash).

**Comment by davidsbailey** [comment]
@islemaster I've added a test. Could you please review that too?

File: shared/middleware/tables_api.rb (hunk 1 above)
Comment (davidsbailey) [inline shared/middleware/tables_api.rb]: Yeah the implicit hash thing is kind of confusing. I think you would only use braces if there were two consecutive hash arguments.

**Comment by islemaster** [comment]
:+1: Looks good!

**Pull request merged**